### PR TITLE
chore: client hooks to validate code and commit messages

### LINKED ISSUE #17: Client hooks to validate code and commit messages
Now that we have the actions to lint commit messages, we should setup client-side hooks to:
- [x] Validate commit message
- [x] Check that the code is correctly formatted

### PR DESCRIPTION
**Pull request by Lumycea**
Closes #17

### CODE REVIEW

**Comment by mochiruntime** [comment]
Thank you for your work! :tada:

In the spirit of not reinventing the wheel and taking advantage of the JS ecosystem, I was thinking about using [commitlint](https://commitlint.js.org/#/) in combination with [Husky](https://typicode.github.io/husky/#/). WDYT about these and how they'd compare against your proposed solution? I think they would integrate better with `npm`-based workflows, and would make it simpler to ensure that the check is consistent across the environment (commitlint is what's already used in #16, and if needed we could also rewrite the PR title check action to use commitlint).

Let me know what you think!

**Review by NoeTerrier — APPROVED** [review]

File: .husky/commit-msg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npx --no -- commitlint --edit "$1"
```
Comment (mochiruntime) [inline .husky/commit-msg:4]: Should we use the `scripts` field of `package.json` instead? That way, we can easily invoke the check with `npm run commitlint`. See my comments in file `package.json`.

```suggestion
npm run commitlint -- --edit "$1"
```

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npx prettier --check .
```
Comment (mochiruntime) [inline .husky/pre-commit:4]: Same as previous comment, see `package.json`

```suggestion
npm run prettier
```

File: commitlint.config.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
+module.exports = {
```
Comment (mochiruntime) [inline commitlint.config.js:1]: Isn't this equivalent to `module.exports = {extends: ['@commitlint/config-conventional']}` ? I think we should only consider "ejecting" only if we need custom a custom config, and I see that you have already installed `@commitlint/config-conventional` in `package.json` so we might as well use it.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+  "scripts": {
+    "prepare": "husky install"
+  },
```
Comment (mochiruntime) [inline package.json:4]:
```suggestion
  "scripts": {
    "check": "npm run commitlint && npm run prettier",
    "commitlint": "commitlint",
    "prettier": "prettier --check .",
    "prepare": "husky install"
  },
```

**Review by mochiruntime — COMMENTED** [review]
Sorry for reviewing so late, since I have some requested changes. Let me know what you think @Thechi2000 and if we change this, we might as well force-push (I will bypass the repo rule if needed).

Note: need to test what I wrote locally, I just edited in my thoughts from Github's PR.

**Pull request merged**